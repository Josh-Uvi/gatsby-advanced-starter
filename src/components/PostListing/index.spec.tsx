import React from "react";
import "jest-styled-components";
import { render } from "@testing-library/react";
import { PostListing } from ".";
import { postListing } from "../../../test/sampleData";

describe("component PostListing", () => {
  it("renders correctly", () => {
    expect.assertions(1);

    const { asFragment } = render(<PostListing listing={postListing} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
