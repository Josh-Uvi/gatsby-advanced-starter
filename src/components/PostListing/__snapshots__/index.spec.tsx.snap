// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component PostListing renders correctly 1`] = `
<DocumentFragment>
  .c20 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c3 {
  object-fit: cover;
  box-shadow: inset 0px 0px 1px 1px rgba(0,0,0,0.1);
  border-radius: 10px;
}

.c16 {
  font-family: "Alfa Slab One",serif;
  font-style: normal;
  font-weight: normal;
  font-size: 26px;
  line-height: 32px;
  -webkit-letter-spacing: 0.15px;
  -moz-letter-spacing: 0.15px;
  -ms-letter-spacing: 0.15px;
  letter-spacing: 0.15px;
}

.c17 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 24px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c12 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  width: 100%;
}

.c4 {
  width: 100%;
  height: 368px;
}

.c1 {
  display: grid;
  grid-gap: 8px;
}

.c21 {
  display: grid;
  grid-gap: 8px;
}

.c7 {
  display: grid;
  grid-gap: 12px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

.c18 {
  display: -webkit-box;
  -webkit-line-clamp: 5;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c23 {
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c6 {
  display: grid;
  grid-gap: 12px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

.c5 {
  display: grid;
}

.c22 {
  display: grid;
}

.c19 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--color-primary);
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 18px;
  line-height: 16px;
  -webkit-letter-spacing: 1.25px;
  -moz-letter-spacing: 1.25px;
  -ms-letter-spacing: 1.25px;
  letter-spacing: 1.25px;
  text-transform: uppercase;
  display: grid;
  grid-gap: 12px;
  grid-auto-flow: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  color: var(--color-primary);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c15 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 10px;
  line-height: 16px;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  text-transform: uppercase;
}

.c15::before {
  content: "#";
}

.c11 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  text-transform: capitalize;
}

.c13 {
  color: var(--color-grey-700);
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c14 {
  display: grid;
  grid-auto-flow: column;
  grid-gap: 16px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 {
  width: 100%;
  display: grid;
  grid-gap: 80px;
  grid-template-columns: 1fr 1fr;
}

@media (min-width:564px) {
  .c16 {
    font-family: "Alfa Slab One",serif;
    font-style: normal;
    font-weight: normal;
    font-size: 33px;
    line-height: 40px;
    -webkit-letter-spacing: 0.15px;
    -moz-letter-spacing: 0.15px;
    -ms-letter-spacing: 0.15px;
    letter-spacing: 0.15px;
  }
}

@media (min-width:564px) {
  .c17 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 28px;
    -webkit-letter-spacing: 0.5px;
    -moz-letter-spacing: 0.5px;
    -ms-letter-spacing: 0.5px;
    letter-spacing: 0.5px;
  }
}

@media (min-width:564px) {
  .c12 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 16px;
    -webkit-letter-spacing: 0.4px;
    -moz-letter-spacing: 0.4px;
    -ms-letter-spacing: 0.4px;
    letter-spacing: 0.4px;
  }
}

@media (max-width:500px) {
  .c4 {
    height: 320px;
  }
}

@media (max-width:440px) {
  .c4 {
    height: 270px;
  }
}

@media (min-width:1100px) {
  .c1 {
    grid-column: span 2;
    grid-template-columns: 6fr 4fr;
    grid-auto-flow: column;
    grid-gap: 32px;
  }
}

@media (min-width:1100px) {
  .c5 {
    -webkit-align-content: space-between;
    -ms-flex-line-pack: space-between;
    align-content: space-between;
  }
}

@media (max-width:1100px) {
  .c19 {
    display: none;
  }
}

@media (min-width:564px) {
  .c15 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 16px;
    -webkit-letter-spacing: 1.5px;
    -moz-letter-spacing: 1.5px;
    -ms-letter-spacing: 1.5px;
    letter-spacing: 1.5px;
    text-transform: uppercase;
  }
}

@media (min-width:564px) {
  .c11 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 16px;
    -webkit-letter-spacing: 0.4px;
    -moz-letter-spacing: 0.4px;
    -ms-letter-spacing: 0.4px;
    letter-spacing: 0.4px;
  }
}

@media (max-width:1100px) {
  .c0 {
    grid-template-columns: 1fr;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        class="c2"
        href="/big-sample-test"
      >
        <img
          class="c3 c4"
          src="5.jpg"
        />
      </a>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <div
                class="c9"
              >
                <a
                  class="c10 c11"
                  href="/categories/moar"
                >
                  moar
                </a>
                <p
                  class="c12 c13"
                >
                   ⋅ Mar 1, 2018 ⋅ 3 min read
                </p>
              </div>
              <div
                class="c14"
              >
                <a
                  class="c10 c15"
                  href="/tags/test"
                >
                  test
                </a>
                <a
                  class="c10 c15"
                  href="/tags/huge"
                >
                  huge
                </a>
              </div>
            </div>
            <a
              class="c2"
              href="/big-sample-test"
            >
              <h3
                class="c16"
              >
                Big Test
              </h3>
            </a>
          </div>
          <a
            class="c2"
            href="/big-sample-test"
          >
            <p
              class="c17 c18"
            >
              MDX Test Here's a color picker! NOTE: This "post" is based on  Markdown Cheatsheet  and is meant to test styling of Markdown generated…
            </p>
          </a>
        </div>
        <a
          class="c19"
          href="/big-sample-test"
        >
          Read the article 
          <svg
            aria-hidden="true"
            class="c20"
            fill="currentColor"
            focusable="false"
            height="28"
            viewBox="0 0 24 24"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.002 3h-14c-1.103 0-2 .897-2 2v4h2V5h14v14h-14v-4h-2v4c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2V5c0-1.103-.898-2-2-2z"
            />
            <path
              d="M11 16l5-4-5-4v3.001H3v2h8z"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      class="c21"
    >
      <a
        class="c2"
        href="/the-fairys-witches"
      >
        <img
          class="c3 c4"
          src="3.jpg"
        />
      </a>
      <div
        class="c22"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <div
                class="c9"
              >
                <a
                  class="c10 c11"
                  href="/categories/something"
                >
                  something
                </a>
                <p
                  class="c12 c13"
                >
                   ⋅ Apr 2, 2017 ⋅ 1 min read
                </p>
              </div>
              <div
                class="c14"
              />
            </div>
            <a
              class="c2"
              href="/the-fairys-witches"
            >
              <h3
                class="c16"
              >
                The Fairy's Witches
              </h3>
            </a>
          </div>
          <a
            class="c2"
            href="/the-fairys-witches"
          >
            <p
              class="c17 c23"
            >
              Eram mora tuas in inter ferens caterva Ait oculos Lorem markdownum. Dedit caput, saxa lenis adhibere negarunt substiterat et ab
testes…
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c21"
    >
      <a
        class="c2"
        href="/the-butterfly-of-the-edge"
      >
        <img
          class="c3 c4"
          src="2.jpg"
        />
      </a>
      <div
        class="c22"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <div
                class="c9"
              >
                <a
                  class="c10 c11"
                  href="/categories/tech"
                >
                  tech
                </a>
                <p
                  class="c12 c13"
                >
                   ⋅ Mar 17, 2017 ⋅ 1 min read
                </p>
              </div>
              <div
                class="c14"
              >
                <a
                  class="c10 c15"
                  href="/tags/programming"
                >
                  programming
                </a>
                <a
                  class="c10 c15"
                  href="/tags/more tags"
                >
                  more tags
                </a>
              </div>
            </div>
            <a
              class="c2"
              href="/the-butterfly-of-the-edge"
            >
              <h3
                class="c16"
              >
                The Butterfly of the Edge
              </h3>
            </a>
          </div>
          <a
            class="c2"
            href="/the-butterfly-of-the-edge"
          >
            <p
              class="c17 c23"
            >
              Hanc indeploratum seram Amphitryoniadae tremulis pauperque manum Cognoscenti audeat ulterius sublimia terga potiorque saxo Lorem markdownum…
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c21"
    >
      <a
        class="c2"
        href="/random-project-4-with-super-long-title-phase-one"
      >
        <img
          class="c3 c4"
          src="4.jpg"
        />
      </a>
      <div
        class="c22"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <div
                class="c9"
              >
                <a
                  class="c10 c11"
                  href="/categories/random"
                >
                  random
                </a>
                <p
                  class="c12 c13"
                >
                   ⋅ Mar 2, 2017 ⋅ 1 min read
                </p>
              </div>
              <div
                class="c14"
              >
                <a
                  class="c10 c15"
                  href="/tags/say hi"
                >
                  say hi
                </a>
              </div>
            </div>
            <a
              class="c2"
              href="/random-project-4-with-super-long-title-phase-one"
            >
              <h3
                class="c16"
              >
                Random project 4 With Super Long Title Phase One
              </h3>
            </a>
          </div>
          <a
            class="c2"
            href="/random-project-4-with-super-long-title-phase-one"
          >
            <p
              class="c17 c23"
            >
              Artificem honorati ore temperat Lycus quam antiquum Mendaci pinu ipsi nunc Lorem markdownum supplex. Care ferre nos praemia detestatur…
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c21"
    >
      <a
        class="c2"
        href="/angels-of-mist"
      >
        <img
          class="c3 c4"
          src="6.jpg"
        />
      </a>
      <div
        class="c22"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <div
                class="c9"
              >
                <a
                  class="c10 c11"
                  href="/categories/test3"
                >
                  test3
                </a>
                <p
                  class="c12 c13"
                >
                   ⋅ Mar 1, 2017 ⋅ 1 min read
                </p>
              </div>
              <div
                class="c14"
              >
                <a
                  class="c10 c15"
                  href="/tags/cheese"
                >
                  cheese
                </a>
                <a
                  class="c10 c15"
                  href="/tags/other"
                >
                  other
                </a>
              </div>
            </div>
            <a
              class="c2"
              href="/angels-of-mist"
            >
              <h3
                class="c16"
              >
                Angels of Mist
              </h3>
            </a>
          </div>
          <a
            class="c2"
            href="/angels-of-mist"
          >
            <p
              class="c17 c23"
            >
              Ab dicemur cura Mane oscula anxia Lorem markdownum alius,  non voluntas supremaque : canes ager cingere quis;
rerum? Nunc nec ferunt verbis…
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c21"
    >
      <a
        class="c2"
        href="/birch-in-the-roses"
      >
        <img
          class="c3 c4"
          src="1.jpg"
        />
      </a>
      <div
        class="c22"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <div
                class="c9"
              >
                <a
                  class="c10 c11"
                  href="/categories/tech"
                >
                  tech
                </a>
                <p
                  class="c12 c13"
                >
                   ⋅ Mar 1, 2017 ⋅ 1 min read
                </p>
              </div>
              <div
                class="c14"
              >
                <a
                  class="c10 c15"
                  href="/tags/tag"
                >
                  tag
                </a>
              </div>
            </div>
            <a
              class="c2"
              href="/birch-in-the-roses"
            >
              <h3
                class="c16"
              >
                Birch in the Roses
              </h3>
            </a>
          </div>
          <a
            class="c2"
            href="/birch-in-the-roses"
          >
            <p
              class="c17 c23"
            >
              Domos primus caelum taedia Accipit alto fecerat mutato centauri haerent dominoque Lorem markdownum nunc fuerat vulgaris ipse. Profecisse…
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c21"
    >
      <a
        class="c2"
        href="/the-fallen-time"
      >
        <img
          class="c3 c4"
          src="7.jpg"
        />
      </a>
      <div
        class="c22"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <div
                class="c9"
              >
                <a
                  class="c10 c11"
                  href="/categories/another one"
                >
                  another one
                </a>
                <p
                  class="c12 c13"
                >
                   ⋅ Feb 1, 2017 ⋅ 1 min read
                </p>
              </div>
              <div
                class="c14"
              >
                <a
                  class="c10 c15"
                  href="/tags/test"
                >
                  test
                </a>
                <a
                  class="c10 c15"
                  href="/tags/something"
                >
                  something
                </a>
              </div>
            </div>
            <a
              class="c2"
              href="/the-fallen-time"
            >
              <h3
                class="c16"
              >
                The Fallen Time
              </h3>
            </a>
          </div>
          <a
            class="c2"
            href="/the-fallen-time"
          >
            <p
              class="c17 c23"
            >
              Autolyci eminus retenta Hoc domum solitos veteremque nostrum Lorem markdownum huc suo ara, dubites celeri mihi bicolor. Secuti non? Suo opus…
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c21"
    >
      <a
        class="c2"
        href="/bold-mage"
      >
        <img
          alt="Bold mage cover"
          class="c3 c4"
          src="8.jpg"
        />
      </a>
      <div
        class="c22"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <div
                class="c9"
              >
                <a
                  class="c10 c11"
                  href="/categories/tech"
                >
                  tech
                </a>
                <p
                  class="c12 c13"
                >
                   ⋅ Jan 1, 2017 ⋅ 1 min read
                </p>
              </div>
              <div
                class="c14"
              >
                <a
                  class="c10 c15"
                  href="/tags/programming"
                >
                  programming
                </a>
                <a
                  class="c10 c15"
                  href="/tags/stuff"
                >
                  stuff
                </a>
              </div>
            </div>
            <a
              class="c2"
              href="/bold-mage"
            >
              <h3
                class="c16"
              >
                Bold Mage
              </h3>
            </a>
          </div>
          <a
            class="c2"
            href="/bold-mage"
          >
            <p
              class="c17 c23"
            >
              Quanto et ius coniunctis urbes Sedisti civiliter Lorem  markdownum  Ixione palus  semper peritura barbaque  in aureus
obliquum erigitur…
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
